src/java/fr/paris/lutece/plugins/workflow/modules/appointment/web/AbstractNotifyAppointmentTaskComponent.java

```diff
@@ -352,4 +390,71 @@ public String doSaveConfig( HttpServletRequest request, Locale locale, ITask tas
 
         return null;
     }
+
+    /**
+     * Get the ID of the Appointment Form selected in the Task's configuration parameters
+     * 
+     * @param request
+     *            The HTTTP request
+     * @param config
+     *            The config of the current Task. Can be null if the config has not been created yet
+     * @return the ID of the Appointment Form selected
+     */
+    private int getSelectedAppointmentFormId( HttpServletRequest request, TaskNotifyAppointmentConfig config )
+    {
+        // When the page is reloaded after a form has been selected
+        String selectedAppointmentForm = request.getParameter( PARAMETER_SELECTED_APPOINTMENT_FORM );
+        if ( StringUtils.isNumeric( selectedAppointmentForm ) )
+        {
+            return Integer.parseInt( selectedAppointmentForm );
+        }
+        if ( config != null )
+        {
+            return config.getIdAppointmentForm( );
+        }
+        return 0;
+    }
+
+    /**
+     * Build and return the URL to modify a task (reloads the page)
+     * 
+     * @param request
+     *            The HTTP request
+     * @param idTask
+     *            The ID of the task whose configuration is getting modified
+     * @param idForm
+     *            ID of the form selected by the user
+     * @return The URL of this config's modification page
+     */
+    private String getUrlToTaskModificationPage( HttpServletRequest request, int idTask, String appointmentForm )
+    {
+        StringBuilder redirectUrl = new StringBuilder( AppPathService.getBaseUrl( request ) );
+        redirectUrl.append( JSP_MODIFY_TASK );
+
+        UrlItem url = new UrlItem( redirectUrl.toString( ) );
+        url.addParameter( PARAMETER_ID_TASK, idTask );
+        url.addParameter( PARAMETER_SELECTED_APPOINTMENT_FORM, appointmentForm );
+
+        return url.getUrl( );
+    }
+
+    /**
+     * Get a ReferenceList containing all the available Appointment Forms
+     * 
+     * @return the ReferenceList of all the Appointment Forms
+     */
+    private ReferenceList getDisplayedFormList( )
+    {
+        // Get a ReferenceList of all available Appointment Forms
+        ReferenceList refListForms = FormService.findAllInReferenceList( );
```

Comment:
We could find the forms associated with the id of the current workflow including the task.
But the problem is that a workflow can be associated to several appointment forms, this is incompatible with this kind of configuration...